### PR TITLE
Feat: list shared

### LINKED ISSUE #53: feat: list shared files
Request `share_list`:

- List files shared with the user

### PR DESCRIPTION
**Pull request by SilviaPabon**
### Includes:
- list shared feat
- list shared feat test
refactors in: 
- move all related tests to ITShareManagement
- method to generate array of files 

This should be close #53

### CODE REVIEW

File: app/src/main/java/gateway/controller/CtrlShareList.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+		String url = Config.getMetadataBaseUrl () + "/files/shared_with_me/" + userUUID;
+
+		try {
+
+			HttpResponse<String> response = HttpClient.newHttpClient ().send (
+				HttpRequest.newBuilder ()
+					.uri (URI.create (url))
+					.GET ()
+					.header ("Authorization", "Bearer " + authorization.token)
```
Comment (Woynert) [inline app/src/main/java/gateway/controller/CtrlShareList.java:34]: Metadata service doesn't receive an authentication header

File: app/src/main/java/gateway/controller/CtrlShareList.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.UUID;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class CtrlShareList
+{
+	public static ResShareList share_list (Authorization authorization)
+	{
+		ResShareList resShareList = new ResShareList ();
+
```
Comment (Woynert) [inline app/src/main/java/gateway/controller/CtrlShareList.java:22]: Missing authentication from auth service

File: app/src/test/java/gateway/ITShareManagement.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+		// List of shared with users successfully
+		assertEquals (
+			200, CtrlShareList.share_list (authorization).code, "Ok. The directory was listed.");
+
+		TestUtilConfig.makeInvalidMetadata ();
+		assertEquals (500, CtrlShareList.share_list (authorization).code, "Can't reach metadata");
+	}
+
+	@Test void shareListWhitWho ()
```
Comment (Woynert) [inline app/src/test/java/gateway/ITShareManagement.java:95]: Somehow this typo passed the previous revision, please rename it to `shareListWithWho`

**Review by Woynert — CHANGES_REQUESTED** [review]
:building_construction:

**Review by Woynert — APPROVED** [review]
:tada: egrem & hsauqS

**Comment by PedroChaparro** [comment]
esaeler eht ekam ot tegrof t'noD

**Comment by SilviaPabon** [comment]
> esaeler eht ekam ot tegrof t'noD

dnatsrednu t'nod I, wow oh

**Comment by Woynert** [comment]
eman dnoces ym s'ti esaeler ,esaelP

**Comment by SilviaPabon** [comment]
😠 sdrawkcab ni yrgna ,fats ym gnilaets llits uoy ,hcnarb nwo ym eteled em tel tnod uoy ,esaeleR trenyoW ko

**Pull request merged**